ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/Auth.kt

```diff
@@ -23,9 +24,20 @@ private class AtomicCounter {
     val atomic = atomic(0)
 }
 
+/**
+ * Configuration used by [Auth] plugin.
+ */
 @KtorDsl
 public class AuthConfig {
+    /**
+     * [AuthProvider] list to use.
+     */
     public val providers: MutableList<AuthProvider> = mutableListOf()
+
+    /**
+     * A lambda function to control whether a response is unauthorized and should trigger a refresh / re-auth.
+     */
+    public var isUnauthorized: suspend (HttpResponse) -> Boolean = { it.status == HttpStatusCode.Unauthorized }
```

Comment:
Let's refine the API a bit to make it more consistent to the existing ones.

```suggestion
    internal var isUnauthorized: suspend (HttpResponse) -> Boolean = { it.status == HttpStatusCode.Unauthorized }

    /** ... */
    public fun isUnauthorized(block: suspend (HttpResponse) -> Boolean) {
        isUnauthorized = block
    }
```

Though, I'm not sure about naming. Probably, `isUnauthorizedResponse` would be better. @marychatte, what do you think?

Reply:
But then you couldn't build on top of existing values:

```kotlin
val original = isUnauthorized
isUnauthorized = { original(it) || ... }
```

Reply:
We could make only `set()` private then, giving the possibility to get the configured `isAuthorized`:
```kotlin
public var isUnauthorized: suspend (HttpResponse) -> Boolean = { it.status == HttpStatusCode.Unauthorized }
    private set()
```
The point is to specify this field without an assignment operator.

Reply:
Why is a getter + a differently named setter function better than an assignment operator unifying those two?

Reply:
Please take a look at the latest commit. I've changed it. Though, this solution seems less direct and understandable, especially when there's a public getter (which probably is missing in many of the other APIs?). The `reAuthorizeOnStatuses` extra function can also still be provided with a var, too, so I don't see that argument.